pandas/core/format.py

```diff
@@ -1895,6 +1903,14 @@ def get_formatted_cells(self):
 
 def format_array(values, formatter, float_format=None, na_rep='NaN',
                  digits=None, space=None, justify='right'):
+
+    if isinstance(values, Index):
```

Comment:
so here add the test for ``is_categorical_dtype``

and just have another part of the if which handles ``PeriodIndex`` (as in ``com.is_period_arraylike``)), as ``datetime64/timedelta64`` are already dispatched correctly.

and tz's are handled in the ``Datetime64Formatter`` (soon the tz-block formattr will handle as well)